resources/js/models/Story.ts

```diff
@@ -0,0 +1,67 @@
+import User from './User';
+import axios from 'axios';
+import fakeStory from './story.json';
+import { config } from '../config';
+import {isToday, isYesterday, format} from 'date-fns';
+
+const mainUrl = `${config.host}/api/stories`;
+export default class Story {
+    id: number;
+    image: {
+        id: number;
+        url: string;
+    };
+    createdAt: string;
+    user: User;
+
+    constructor(json: {
+        id: number;
+        image: {
+            id: number;
+            url: string;
+        };
+        createdAt: string;
+        user: User;
+    }) {
+        this.id = json.id;
+        this.image = json.image;
+        this.createdAt = json.createdAt;
+        this.user = json.user;
+    }
+
+    static fromJson(json: any): Story {
+        return new Story(json);
+    }
+
+    static async all(): Promise<Story[]> {
+            try{
+                return fakeStory.map((story) => Story.fromJson(story));
+            }
+            catch{
+                throw new Error('Error fetching users')
+            }
+    }
+
+      getFormattedDate = () => {
+        if (isToday(this.createdAt)) {
+            return `Toady at ${format((this.createdAt), 'HH:mm')}`
+        }
+        if (isYesterday(this.createdAt)) {
+            return `Yesterday at ${format((this.createdAt), 'HH:mm')}`
+        }
+        return format((this.createdAt), 'dd/MM/yyyy')
+    };
+
+    getFormattedStory = () => {
+        return {
+            id: this.id,
+            imageUrl: this.image.url || 'https://picsum.photos/200/300',
+            duration: 5000,
```

Comment:
you have also mixed fetched data with design data (duration) which is a NONO!

in summery there is a lot of wrong things here you should change the way you have implemented components, so you not relay on such 'factory'

what you have done here is known as 'factory method' which in this case is a bad use case because you wasn't able to use model itself so you implemented a method to abstract away the original model structure and gives you what you actually need in your components.

this will result in a bigger mess later, so lets fix it now and never use such thing agin

Reply:
I didnt understand how should write this part 😅